system/lib/libcxxabi/src/cxa_exception.cpp

```diff
@@ -644,6 +644,14 @@ void __cxa_rethrow() {
     }
 #ifdef __USING_SJLJ_EXCEPTIONS__
     _Unwind_SjLj_RaiseException(&exception_header->unwindHeader);
+#elif __USING_WASM_EXCEPTIONS__
+#ifdef NDEBUG
+    _Unwind_RaiseException(&exception_header->unwindHeader);
+#else
+    // In debug mode, call a JS library function to use WebAssembly.Exception JS
+    // API, which enables us to include stack traces
+    __throw_exception_with_stack_trace(&exception_header->unwindHeader);
```

Comment:
This makes me wonder why `_Unwind_RaiseException` itself doesn't call `__throw_exception_with_stack_trace` under the hood?   I'm sure there is a good reason..

Reply:
I have a reason, even though how good that reason is... In native platforms libunwind is mostly used for very low level stack unwinding stuff, so I wanted to limit libunwind to low-level stuff as well, such as throwing exceptions using builtins. Interacting with JS seemed a too high level functionality for libunwind.

Reply:
I guess it depends if we think there would be other callers of ` _Unwind_RaiseException` that don't want `__throw_exception_with_stack_trace` ?

Reply:
libunwind can be used by non-Emscripten toolchains, so there are others that don't want to call it. But that's gonna be the same for libc++abi... Maybe we should guard this part as not only `__USING_WASM_EXCEPTIONS__` but also `__EMSCRIPTEN__`? Currently I haven't upstreamed any JS-related parts, including this, to upstream LLVM, so others will not be affected though.

Reply:
Yes I think it makes sense to wrap this in `__EMSCRIPTEN__`.    We can consider folding this into `_Unwind_RaiseException` itself, either as part of this PR, or as a followup.

Reply:
Also, I don't see any harm in (eventually) upstreaming even the emscripten parts, assuming they are wrapped in `#ifdef __EMSCRIPTEN__`